cpmpy/solvers/pysat.py

```diff
@@ -224,6 +235,9 @@ def _post_constraint(self, cpm_expr):
             Post a primitive CPMpy constraint to the native solver API
         """
         from pysat.card import CardEnc
+        if not CPM_pysat.pb_supported():
```

Comment:
is this not only in case 'PB' is used?

i would not want users not to be able to solve plain SAT models if they want to, because they don't have a library installed they dont want to use.


Also, it looks like you are completely replacing CardEnc by PBenc, would it not make sense to have both, and only use PBenc in case of a wsum?

Reply:
Adapted the PBenc to Cardenc (regular sum) and PBEnc (wsum)